In [1]:
import pandas as pd
import requests
import json

# import api key from untracked - config.py
from config import nps_api_key

### Try out API Queries to NPS dataset
____________

## Parks data

In [2]:
#  limit is arbitrary value greater than total number of NPS sites 
limit = 800 
content = "parks"
endpoint = "https://developer.nps.gov/api/v1/%s?limit=%s&api_key="%(content, limit)

response = requests.get(endpoint + nps_api_key)
parks_overview_json = response.json()

parks_df = pd.DataFrame.from_records(parks_overview_json["data"])

In [5]:
parks_df.head(3)

id                                 url  \
0  77E0D7F0-1942-494A-ACE2-9004D2BDC59E  https://www.nps.gov/abli/index.htm   
1  6DA17C86-088E-4B4D-B862-7C1BD5CF236B  https://www.nps.gov/acad/index.htm   
2  E4C7784E-66A0-4D44-87D0-3E072F5FEF43  https://www.nps.gov/adam/index.htm   

                                            fullName parkCode  \
0  Abraham Lincoln Birthplace National Historical...     abli   
1                               Acadia National Park     acad   
2                     Adams National Historical Park     adam   

                                         description    latitude  \
0  For over a century people from around the worl...  37.5858662   
1  Acadia National Park protects the natural beau...   44.409286   
2  From the sweet little farm at the foot of Penn...  42.2553961   

      longitude                            latLong  \
0  -85.67330523  lat:37.5858662, long:-85.67330523   
1    -68.247501     lat:44.409286, long:-68.247501   
2  -71.01160356  lat:42.2553961, long:-71.01160356   

                                          activities  \
0  [{'id': '13A57703-BB1A-41A2-94B8-53B692EB7238'...   
1  [{'id': '09DF0950-D319-4557-A57E-04CD2F63FF42'...   
2  [{'id': 'B33DC9B6-0B7D-4322-BAD7-A13A34C584A3'...   

                                              topics  ...  \
0  [{'id': 'D10852A3-443C-4743-A5FA-6DD6D2A054B3'...  ...   
1  [{'id': '00F3C3F9-2D67-4802-81AE-CCEA5D3BA370'...  ...   
2  [{'id': 'F3883A66-A7CB-461B-868E-1B5932224B25'...  ...   

                                      entrancePasses fees  \
0                                                 []   []   
1  [{'cost': '55.00', 'description': 'Valid for 1...   []   
2                                                 []   []   

                                      directionsInfo  \
0  The Birthplace Unit of the park is located app...   
1  From Boston take I-95 north to Augusta, Maine,...   
2  Traveling on U.S. Interstate 93, take exit 7 -...   

                                       directionsUrl  \
0  http://www.nps.gov/abli/planyourvisit/directio...   
1  http://www.nps.gov/acad/planyourvisit/directio...   
2  http://www.nps.gov/adam/planyourvisit/directio...   

                                      operatingHours  \
0  [{'exceptions': [{'exceptionHours': {}, 'start...   
1  [{'exceptions': [], 'description': 'In support...   
2  [{'exceptions': [{'exceptionHours': {'wednesda...   

                                           addresses  \
0  [{'postalCode': '42748', 'city': 'Hodgenville'...   
1  [{'postalCode': '04609', 'city': 'Bar Harbor',...   
2  [{'postalCode': '02169', 'city': 'Quincy', 'st...   

                                              images  \
0  [{'credit': 'NPS Photo', 'title': 'The Memoria...   
1  [{'credit': 'NPS / Kristi Rugg', 'title': 'Sun...   
2  [{'credit': 'NPS Photo', 'title': 'The John an...   

                                         weatherInfo  \
0  There are four distinct seasons in Central Ken...   
1  Located on Mount Desert Island in Maine, Acadi...   
2  Be prepared for hot, humid weather. The histor...   

                         name               designation  
0  Abraham Lincoln Birthplace  National Historical Park  
1                      Acadia             National Park  
2                       Adams  National Historical Park  

[3 rows x 23 columns]

In [9]:
parks_df.weatherInfo.values[0]

'There are four distinct seasons in Central Kentucky. However, temperature and weather conditions can vary widely within those seasons. Spring and Fall are generally pleasant with frequent rain showers. Summer is usually hot and humid. Winter is moderately cold with mixed precipitation.'

In [11]:
parks_df.weatherInfo.values[-1]

'Zion is known for a wide range of weather conditions. Temperatures vary with changes in elevation and day/night temperatures may differ by over 30°F. In summer, temperatures in Zion National Park often exceed 100°F/38°C. Zion experiences monsoons from mid-July into September that results in an increased risk of flash floods. Always be aware of the threat of storms and lightning and be prepared for a wide range of weather conditions. Winters are generally mild.'

____
## Parks table: SUMMARY
- general main table for list of national parks 
- info from this table will look good on the front end
- use lat/long to make a pretty map
____

## Things to Do data

In [69]:
limit = 800 
content = "thingstodo"
endpoint = "https://developer.nps.gov/api/v1/%s?limit=%s&api_key="%(content, limit)

response = requests.get(endpoint + nps_api_key)
json_data = response.json()

todo_df = pd.DataFrame.from_records(json_data["data"])

In [70]:
todo_df.head(3)

id  \
0  F7B724F1-73D2-4266-B9B5-ED69270FAFF1   
1  DDB3F3CD-A710-4144-B28D-6B7006A8D0E6   
2  D662DD28-8731-455E-B59D-FC6923BFC596   

                                                 url  \
0  https://www.nps.gov/thingstodo/explore-with-th...   
1  https://www.nps.gov/thingstodo/take-a-guide-to...   
2  https://www.nps.gov/thingstodo/take-a-self-gui...   

                                  title  \
0      Explore With the Cell Phone Tour   
1       Take a Guided Tour at Weir Farm   
2  Take a Self-Guided Tour of Weir Farm   

                                    shortDescription  \
0  The Oncell Cell Phone tour is a great way to e...   
1  Tour the landscape that inspired three generat...   
2  Stroll through gardens and grounds, complete w...   

                                              images  \
0  [{'url': 'https://www.nps.gov/common/uploads/c...   
1  [{'url': 'https://www.nps.gov/common/uploads/c...   
2  [{'url': 'https://www.nps.gov/common/uploads/c...   

                                        relatedParks relatedOrganizations  \
0  [{'states': 'KS', 'parkCode': 'fols', 'designa...                   []   
1  [{'states': 'CT', 'parkCode': 'wefa', 'designa...                   []   
2  [{'states': 'CT', 'parkCode': 'wefa', 'designa...                   []   

                                                tags  latitude   longitude  \
0  [Outdoor activities, fort larned, indian wars,...  38.18372  -99.218562   
1  [art, art studio, painting, Guided Tour, range...                         
2  [Self-guided Tour, tour, self guided, Weir Far...                         

   ... locationDescription doFeesApply  \
0  ...                           false   
1  ...                           false   
2  ...                           false   

                                     longDescription  \
0  <p>The Oncell Cell Phone tour is a great way t...   
1  <p>Tour the landscape that inspired three gene...   
2  <p>Stroll through gardens and grounds, complet...   

                              reservationDescription  \
0                                                      
1  <p>Parking is limited!</p> <p>Please call ahea...   
2  <p>Parking is limited!</p> <p>Please call ahea...   

                           season  \
0  [Winter, Spring, Summer, Fall]   
1          [Spring, Summer, Fall]   
2  [Winter, Spring, Summer, Fall]   

                                              topics durationDescription  \
0  [{'id': '28AEAE85-9DDA-45B6-981B-1CFCDCC61E14'...                       
1  [{'id': '00F3C3F9-2D67-4802-81AE-CCEA5D3BA370'...                       
2  [{'id': '00F3C3F9-2D67-4802-81AE-CCEA5D3BA370'...                       

  arePetsPermitted     timeOfDay        duration  
0             true         [Day]   45-60 Minutes  
1             true         [Day]   10-60 Minutes  
2             true  [Dawn, Dusk]  30-120 Minutes  

[3 rows x 33 columns]

In [71]:
all_tags = [inner for outer in todo_df['tags'].values for inner in outer]

In [72]:
len(set(all_tags))

1596

In [73]:
all_tags.count('hiking')

183

In [74]:
all_tags.count('hike')

36

In [83]:
hikes_df = todo_df[todo_df['tags'].apply(lambda x: 'hike' in x 
                                      or 'hiking' in x 
                                      or 'trail' in x
                                      or 'trails' in x)]

In [86]:
hikes_df.iloc[0]

id                                               F767FB48-D6BF-4EE7-8F89-6499246BE441
url                                 https://www.nps.gov/thingstodo/hike-the-vc-loo...
title                               Hike the Blue Ridge Parkway Visitor Center Loo...
shortDescription                    The Blue Ridge Parkway Visitor Center Loop Tra...
images                              [{'url': 'https://www.nps.gov/common/uploads/c...
relatedParks                        [{'states': 'NC,VA', 'parkCode': 'blri', 'desi...
relatedOrganizations                                                               []
tags                                [hiking, trail, track trail, forested trail, f...
latitude                                                           35.564776566271604
longitude                                                          -82.48728275044944
amenities                                                                          []
location                              Blue Ridge Parkw

In [87]:
hikes_df.shape

(225, 33)

____
## Things To Do table: SUMMARY (todos)
- Is there a better way to get hikes?
    - seems like nat park system should have more than 225 hikes
    - are there hikes hiding under other tags or not tagged 
      - case sensitivity etc
- Link back to parks df 
- Season info?
- other fields potentially useful depending on project

___